adapters/conversant/conversant.go

```diff
@@ -1,291 +1,177 @@
-		if params.SiteID != "" {
-			if cnvrReq.Site != nil {
-				cnvrReq.Site.ID = params.SiteID
-			}
-			if cnvrReq.App != nil {
-				cnvrReq.App.ID = params.SiteID
-			}
+		if cnvrExt.SiteID == "" {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Impression[%d] requires ext.bidder.site_id", i),
+			}}
 		}
 
-		if params.Mobile != nil && !(cnvrReq.Site == nil) {
-			cnvrReq.Site.Mobile = *params.Mobile
-		}
-
-		// Fill in additional impression info
-
-		imp.DisplayManager = "prebid-s2s"
-		imp.DisplayManagerVer = "1.0.1"
-		imp.BidFloor = params.BidFloor
-		imp.TagID = params.TagID
-
-		var position *openrtb.AdPosition
-		if params.Position != nil {
-			position = openrtb.AdPosition(*params.Position).Ptr()
+		if i == 0 {
+			if request.Site != nil {
+				request.Site.ID = cnvrExt.SiteID
+			} else if request.App != nil {
+				request.App.ID = cnvrExt.SiteID
+			}
 		}
+		parseCnvrParams(&request.Imp[i], cnvrExt)
+	}
 
-		if imp.Banner != nil {
-			imp.Banner.Pos = position
-		} else if imp.Video != nil {
-			imp.Video.Pos = position
+	//create the request body
+	data, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error in packaging request to JSON"),
+		}}
+	}
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     c.URI,
+		Body:    data,
+		Headers: headers,
+	}}, errors
+}
 
-			if len(params.API) > 0 {
-				imp.Video.API = make([]openrtb.APIFramework, 0, len(params.API))
-				for _, api := range params.API {
-					imp.Video.API = append(imp.Video.API, openrtb.APIFramework(api))
-				}
-			}
+func parseCnvrParams(imp *openrtb.Imp, cnvrExt openrtb_ext.ExtImpConversant) {
+	imp.DisplayManager = "prebid-s2s"
+	imp.DisplayManagerVer = "2.0.0"
+	imp.BidFloor = cnvrExt.BidFloor
+	imp.TagID = cnvrExt.TagID
 
-			// Include protocols, mimes, and max duration if specified
-			// These properties can also be specified in ad unit's video object,
-			// but are overridden if the custom params object also contains them.
+	// Take care not to override the global secure flag
+	if (imp.Secure == nil || *imp.Secure == 0) && cnvrExt.Secure != nil {
+		imp.Secure = cnvrExt.Secure
+	}
 
-			if len(params.Protocols) > 0 {
-				imp.Video.Protocols = make([]openrtb.Protocol, 0, len(params.Protocols))
-				for _, protocol := range params.Protocols {
-					imp.Video.Protocols = append(imp.Video.Protocols, openrtb.Protocol(protocol))
-				}
-			}
+	var position *openrtb.AdPosition
+	if cnvrExt.Position != nil {
+		position = openrtb.AdPosition(*cnvrExt.Position).Ptr()
+	}
+	if imp.Banner != nil {
+		imp.Banner.Pos = position
 
-			if len(params.MIMEs) > 0 {
-				imp.Video.MIMEs = make([]string, len(params.MIMEs))
-				copy(imp.Video.MIMEs, params.MIMEs)
-			}
+	} else if imp.Video != nil {
+		imp.Video.Pos = position
```

Comment:
`imp.Video` is shared memory. Please make a shallow copy first (see code sample above for .Site).